test/cpp/api/clip_grad.cpp

```diff
@@ -0,0 +1,110 @@
+#include <gtest/gtest.h>
+
+#include <torch/torch.h>
+
+#include <test/cpp/api/support.h>
+
+#include <torch/nn/utils/clip_grad.h>
+
+using namespace torch::nn;
+using namespace torch::test;
+
+struct ClipGradTest : torch::test::SeedingFixture {};
+
+class TestLinearModel : public torch::nn::Module {
+ public:
+  TestLinearModel() : l1(register_module("l1", Linear(10, 10))) {}
+  Linear l1;
+};
+
+TEST_F(ClipGradTest, ClipGrad) {
+  TestLinearModel m;
+  auto linear_layer = m.l1;
+  float max_norm = 2;
+  auto compute_norm = [linear_layer](float norm_type) -> float {
+    float inf = std::numeric_limits<float>::infinity();
+    float total_norm = 0.0;
+    if (norm_type != inf) {
```

Comment:
Since `inf` is only used once, we can write
```suggestion
    if (norm_type != std::numeric_limits<float>::infinity()) {
```